compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1747,11 +1744,18 @@ impl<'a> Parser<'a> {
             body = self.mk_expr_err(body.span);
         }
 
-        Ok(self.mk_expr(
+        let body_span = body.span;
+
+        let clo = self.mk_expr(
```

Comment:
Can we rename this to either `closure` or `expr`? `clo` isn't used anywhere else and might lead to short lived confusion, but confusion nonetheless.